google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -792,6 +792,50 @@ private static LogName getLogName(
     return destination.toLogName(logName);
   }
 
+  public Iterable<LogEntry> populateMetadata(
+      Iterable<LogEntry> logEntries,
+      MonitoredResource customResource,
+      String... exclusionClassPaths) {
+    final Boolean needDebugInfo =
```

Comment:
Do we need to validate `logEntries` for null first?

Reply:
Will do.